certora/D3MHub.spec

```diff
@@ -1,76 +1,76 @@
-    mint(address, uint256) => DISPATCHER(true)
+    function vat() external returns (address) envfree;
+    function daiJoin() external returns (address) envfree;
+    function vow() external returns (address) envfree;
+    function end() external returns (address) envfree;
+    function ilks(bytes32) external returns (address, address, uint256, uint256, uint256) envfree;
+    function locked() external returns (uint256) envfree;
+    function plan(bytes32) external returns (address) envfree;
+    function pool(bytes32) external returns (address) envfree;
+    function tic(bytes32) external returns (uint256) envfree;
+    function tau(bytes32) external returns (uint256) envfree;
+    function culled(bytes32) external returns (uint256) envfree;
+    function wards(address) external returns (uint256) envfree;
+    function vat.can(address, address) external returns (uint256) envfree;
+    function vat.debt() external returns (uint256) envfree;
+    function vat.dai(address) external returns (uint256) envfree;
+    function vat.gem(bytes32, address) external returns (uint256) envfree;
+    function vat.Line() external returns (uint256) envfree;
+    function vat.live() external returns (uint256) envfree;
+    function vat.ilks(bytes32) external returns (uint256, uint256, uint256, uint256, uint256) envfree;
+    function vat.sin(address) external returns (uint256) envfree;
+    function vat.urns(bytes32, address) external returns (uint256, uint256) envfree;
+    function vat.vice() external returns (uint256) envfree;
+    function vat.wards(address) external returns (uint256) envfree;
+    function dai.allowance(address, address) external returns (uint256) envfree;
+    function dai.balanceOf(address) external returns (uint256) envfree;
+    function dai.totalSupply() external returns (uint256) envfree;
+    function dai.wards(address) external returns (uint256) envfree;
+    function daiJoin.dai() external returns (address) envfree;
+    function daiJoin.live() external returns (uint256) envfree;
+    function daiJoin.vat() external returns (address) envfree;
+    function end.debt() external returns (uint256) envfree;
+    function end.gap(bytes32) external returns (uint256) envfree;
+    function end.tag(bytes32) external returns (uint256) envfree;
+    function end.vat() external returns (address) envfree;
+    function end.vow() external returns (address) envfree;
+    function plan.dai() external returns (address) envfree;
+    function pool.hub() external returns (address) envfree;
+    function pool.vat() external returns (address) envfree;
+    function pool.dai() external returns (address) envfree;
+    function pool.share() external returns (address) envfree;
+    function share.balanceOf(address) external returns (uint256) envfree;
+    function share.totalSupply() external returns (uint256) envfree;
+    function share.wards(address) external returns (uint256) envfree;
+    function _.debt() external => DISPATCHER(true);
```

Comment:
I'm doing some tests, but it seems a bunch of these ones can be removed now that is possible to link structures.

Reply:
feel free to submit commits

Reply:
feel free to submit commits